README.rst

```diff
@@ -1,67 +1,69 @@
-It takes about 10 minutes for the build to occur, after pushing to master;
-the status is available `here <https://registry.hub.docker.com/u/clusterhq/build.clusterhq.com/builds_history/46090/>`_).
+The code uses Fabric to start and manage the Buildbot master.
 
-The production instance is accessed using a key from https://github.com/hybridlogic/HybridDeployment (this repository is not publicly available).
-Add the HybridDeployment master key to your authentication agent::
+To install dependencies::
 
-   $ ssh-add /path/to/HybridDeployment/credentials/master_key
+   $ pip install pyyaml
+   $ pip install fabric
 
-Go to a checkout of the build.clusterhq.com repository.
 
+Create the configuration
+------------------------
 
-Install dependencies::
+The configuration requires secret data that must not be committed to the Github repository.
+The secret data is provided in a file ``config.yml``.
 
-   $ pip install pyyaml
-   $ pip install fabric
+A sample configuration is provided in the file ``config.yml.sample``.
 
-The secrets for the master must be stored in a file ``config.yml`` inside the checkout.
-Fabric passes these variables to buildbot via a docker environment variable as JSON.
-These secrets can be found in the "config@build.clusterhq.com" file in LastPass.
-If you have the lastpass command-line client installed, you can use fabric to download the current config::
+The config file for the ClusterHQ master can be found in the ``config@build.clusterhq.com`` file in LastPass.
+If you have the ``lastpass`` command-line client installed, you can use Fabric to download the current config::
 
    $ fab getConfig
 
-Check if anyone has running builds at http://build.clusterhq.com/buildslaves.
-
-Announce on Zulip's Engineering > buildbot stream that Buildbot will be unavailable for a few minutes.
+Fabric passes these variables to Buildbot via a Docker environment variable as JSON.
 
-Update the live Buildbot to the latest image (this may take some time)::
 
-   $ fab update
+Test changes on staging server
+------------------------------
 
-To view the logs::
+Changes to the Buildbot master can be tested on a staging machine.
 
-   $ fab logs
+Create a staging Docker image
+=============================
 
-To restart the live Buildbot with the current image::
+To create a new staging image in the Docker registry, update the ``staging`` branch and push to Github.
+The Docker registry will automatically build an image based on the ``staging`` branch of https://github.com/ClusterHQ/build.clusterhq.com whenever it is updated.
+To make the ``staging`` branch the same as a development branch, run the following commands::
 
-   $ fab restart
+   git checkout staging
+   git pull
+   git reset --hard <other-branch>
+   git reset --soft HEAD@{1}
+   git commit
+   git push
 
-Staging changes
----------------
+After pushing a change to ``staging``, it takes about 10 minutes for the Docker image build to finish.
```

Comment:
Add a note about the permissions necessary to view error reporting at the registry.